generators/readme/src/index.ts

```diff
@@ -0,0 +1,170 @@
+  type: 'text',
+  value,
+});
+const paragraph = (value: string): Paragraph => ({
+  type: 'paragraph',
+  children: [text(value)],
+});
+const heading = (depth: Heading['depth'], value: string): Heading => ({
+  type: 'heading',
+  depth,
+  children: [text(value)],
+});
+const code = (lang: string, value: string): Code => ({
+  type: 'code',
+  lang,
+  value,
+});
+const parseMarkdownFragmentToAST = (filePath: string) => {
+  const fragmentFile = toVfile.readSync(filePath);
+  const fragmentAST = unified()
+    .use(markdown)
+    .use(stringify, stringfyOptions)
+    .use(mdx)
+    .parse(fragmentFile) as Root;
+
+  return fragmentAST.children;
+};
+
+function readmeTransformer(options: GeneratorReadmeOptions) {
+  const packageJsonRaw = fs.readFileSync(
+    path.join(options.packagePath, 'package.json'),
+    {
+      encoding: 'utf8',
+    }
+  );
+  const parsedPackageJson = JSON.parse(packageJsonRaw);
+  const packageJsonInfo: PackgeJsonInfo = {
+    name: parsedPackageJson.name,
+    description: parsedPackageJson.description,
+    version: parsedPackageJson.version,
+    peerDependencies: parsedPackageJson.peerDependencies,
+  };
+
+  return transformer;
+
+  function transformer(tree: Node, _file: VFile) {
+    (tree as Parent).children = [
+      html(
+        [
+          '<!-- THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. -->',
+          '<!-- This file is created by the `yarn generate-readme` script. -->',
+        ].join('\n')
+      ),
+      // The main heading is derived by the name of the folder
+      heading(1, capitalize(camelcase(path.basename(options.packagePath)))),
+      // The description is what's defined in the package.json
+      heading(2, 'Description'),
+      paragraph(packageJsonInfo.description),
+      // Describe the installation steps
+      heading(2, 'Installation'),
+      code('', `yarn add ${packageJsonInfo.name}`),
+      code('', `npm --save install ${packageJsonInfo.name}`),
+      packageJsonInfo.peerDependencies &&
+        paragraph(
+          'Additionally install the peer dependencies (if not present)'
+        ),
+      packageJsonInfo.peerDependencies &&
+        code(
+          '',
+          `yarn add ${Object.keys(packageJsonInfo.peerDependencies).join(' ')}`
+        ),
+      code(
+        '',
+        `npm --save install ${Object.keys(
+          packageJsonInfo.peerDependencies
+        ).join(' ')}`
+      ),
+      // Usage example
+      heading(2, 'Usage'),
+      code(
+        'jsx',
+        fs.readFileSync(
+          path.join(options.packagePath, 'docs/usage-example.js'),
+          { encoding: 'utf8' }
+        )
+      ),
+      // Describe the component's properties
+      heading(2, 'Properties'),
+      ...parseMarkdownFragmentToAST(
+        path.join(options.packagePath, 'docs/properties.md')
+      ),
+      ...parseMarkdownFragmentToAST(
+        path.join(options.packagePath, 'docs/additional-info.md')
+      ),
+    ];
+  }
```

Comment:
This is the most important part.

Reply:
Nice work. Just two things that cross my mind,
1-  What happens if the file is present but empty. IMO we should show that section
2- What happens if the file is not present. Wouldn't this throw errors while fetching the file

Reply:
Which file are you referring to? The fragments?

If so, some of them should be required. But you're right, the optional ones (like the `additional-info`) can be optional, so we should check for that.
Thanks

Reply:
> If so, some of them should be required

do we show warning or throw error incase they are missing, i haven't noticed, sorry.
It is easy to forget adding the required files but showing a warning or error would be helpful

Reply:
Yes sure. There is no validation built-in yet but we can add it for sure 👌